homeassistant/components/dsmr/const.py

```diff
@@ -242,41 +242,17 @@
         entity_category=EntityCategory.DIAGNOSTIC,
     ),
     DSMRSensorEntityDescription(
-        key=obis_references.LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL,
```

Comment:
This is quite a heavy breaking change for long term statistic and people using energy monitoring feature.

These people will loose their data and history.

How can we avoid this being breaking?

Reply:
What is the relation between these changes and the bump to 0.32?
As far as the changelog goes, the only thing changed is support for reading P1 over RFXtrx.

Reply:
> This is quite a heavy breaking change for long term statistic and people using energy monitoring feature.
>  
> These people will loose their data and history.
> 
> How can we avoid this being breaking?

Actually, I do not think it is a breaking change. The constants got renamed, but the underlying string values are the same.
The fact that these were duplicates was the reason to refactor those constants (disclaimer, I did not make those changes myself).

See https://github.com/ndokter/dsmr_parser/issues/95 and https://github.com/ndokter/dsmr_parser/commit/83ef354c1285def5beecbd077670971e74b388e3.

Reply:
> What is the relation between these changes and the bump to 0.32? As far as the changelog goes, the only thing changed is support for reading P1 over RFXtrx.

As far as I can tell, these changes were done after version 31 was released, but never published as a new package version by the package maintainer.
So, in order for me to do the necessary changes to support the P1 over RFXtrx (which I plan to send in a separate PR) I need to include these updates in HA first.

Reply:
> Actually, I do not think it is a breaking change. 

You've removed sensors in this PR, that is a breaking change from my perspective.

Reply:
You are probably right. Let me try to better understand what's going on then:

As far as I understood the code, there are two entity descriptions with the same key (one for Luxembourg and one for Sweden), that get added depending on the dsmr_version.
In essence, they create the same entity if the dsmr_version matches.

In the code as I changed it, it still creates the entity description with the same key (but using the new constant) by including "5L" and "5S" in the `dsmr_version` list for the `ELECTRICITY_IMPORTED_TOTAL` and `ECTRICITY_EXPORTED_TOTAL` constants, respectively.

Does that mean that there will be a different entity created in HA after all when running this code?

Reply:
W.r.t. this change, after some digging, I think as well this does not break anything, but just cleans up.

We had for "5L":
`obis_references.LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL`
`obis_references.LUXEMBOURG_ELECTRICITY_DELIVERED_TARIFF_GLOBAL`
And for "5S":
`obis_references.SWEDEN_ELECTRICITY_USED_TARIFF_GLOBAL`
`obis_references.SWEDEN_ELECTRICITY_DELIVERED_TARIFF_GLOBAL`
And for "2.2" "4" "5" and "5B":
`obis_references.ELECTRICITY_IMPORTED_TOTAL`

So basically here: https://github.com/ndokter/dsmr_parser/pull/96
It was cleaned up because:
`obis_references.LUXEMBOURG_ELECTRICITY_USED_TARIFF_GLOBAL`==`obis_references.SWEDEN_ELECTRICITY_USED_TARIFF_GLOBAL`==`obis_references.ELECTRICITY_IMPORTED_TOTAL`
and
`obis_references.LUXEMBOURG_ELECTRICITY_DELIVERED_TARIFF_GLOBAL`==`obis_references.SWEDEN_ELECTRICITY_DELIVERED_TARIFF_GLOBAL`

For the latter, a new key `obis_references.ELECTRICITY_EXPORTED_TOTAL` has been introduced.

Because the key values behind it were identical, cleaning it up this way would not break something.

Reply:
Thanks for explaining this better than I did

Reply:
From a library point of view, you guys are absolutely right. From the Home Assistant end, unfortunately not.

The existing sensors define their unique ID based on the name of the entity description:

```py
        self._attr_unique_id = f"{device_serial}_{entity_description.name}".replace(
            " ", "_"
        )
```

Thus removing an entity description, loses a sensor and unique reference (and thus cases orphaned entities and loss of data as I already mentioned above).

My review comment remains the same for that reason.

Reply:
But the names are identical either right?

Reply:
So in case of consumption, previously there were 3 listings, all had name `Energy Consumption (total)`, the only differencs was obis key, which turns out to be the same. Only thing changed is that it's now one entry for all versions pointing to the same obis key. Name was already similar as far as I can see.

Reply:
😵‍💫  Absolutely right. That is actually awesome!

In that case, I think this is good to go 👍 

Sorry for the confusion there @rhpijnacker!